### PR TITLE
privatize a few methods + cache hashes instead of entries

### PR DESCRIPTION
**Pull request by grosser**
@byroot

### CODE REVIEW

File: lib/bootscale/file_storage.rb [hunk 1]
```diff
@@ -13,11 +13,13 @@ def load(load_path)
 
     def dump(load_path, cache)
       path = cache_path(load_path)
-      return if File.exist?(path)
+      raise "This should not happen" if File.exist?(path)
```
Comment (byroot) [inline lib/bootscale/file_storage.rb]: Yes it would happen often.

Comment (grosser) [inline lib/bootscale/file_storage.rb]: I though dump only gets called when load did not work, so that means the path did not exist ?

Comment (byroot) [inline lib/bootscale/file_storage.rb]: Hum, good point. What I just said is not up to date. :+1: but we'd need to assert it via another integration test.

Comment (grosser) [inline lib/bootscale/file_storage.rb]: added test + made tests more reliable by running them in tmpdir instead of global /tmp

Comment (grosser) [inline lib/bootscale/file_storage.rb]: otherwise all looking good ?

Comment (byroot) [inline lib/bootscale/file_storage.rb]: Yes, but I think I would simply remove this line. If it's never gonna happen anymore, we might as well not do a useless `stat` call :)

Comment (grosser) [inline lib/bootscale/file_storage.rb]: true true ... I'll change the test to 'it ca dump twice' or jut drop it ?

Comment (byroot) [inline lib/bootscale/file_storage.rb]: Just drop it I guess.

Comment (grosser) [inline lib/bootscale/file_storage.rb]: done!

**Pull request merged**